packages/cli/src/__tests__/generator.test.ts

```diff
@@ -0,0 +1,476 @@
+import { execFileSync } from "child_process"
+import * as dedent from "dedent"
+import * as fs from "fs"
+import * as path from "path"
+import {
+  ArrowFunction,
+  AwaitExpression,
+  BodyableNodeStructure,
+  CallExpression,
+  ExpressionStatement,
+  JsxSelfClosingElement,
+  MethodDeclaration,
+  ObjectLiteralExpression,
+  Project,
+  PropertyAssignment,
+  SourceFile,
+  StringLiteral,
+  SyntaxList,
+  VariableStatement,
+} from "ts-simple-ast"
+import { run as runGenerator } from "yeoman-test"
+import ComponentGenerator = require("../generator")
+
+function compileGenerator() {
+  try {
+    execFileSync(
+      process.argv[0],
+      [
+        path.resolve(__dirname, "../../../../node_modules/.bin/tsc"),
+        "--target",
+        "es6",
+        "--module",
+        "commonjs",
+        path.join(__dirname, "../generator/index.ts"),
+      ],
+      { env: process.env },
+    )
+  } catch (_) {
+    // no-op, the error thrown is type errors, which we don’t care about here.
+  }
+  return path.join(__dirname, "../generator/index.js")
+}
+
+function generate(component: string, options = {}) {
+  const context = runGenerator(path.join(__dirname, "../generator"))
+    .withOptions(options)
+    .withArguments([component])
+
+  let error: Error
+  context.on("ready", (generator: any) => {
+    generator.env.on("error", (err: Error) => {
+      error = err
+    })
+  })
+
+  return context.then(() => {
+    if (error) {
+      throw error
+    }
+    const project = new Project({})
+    const sourceFile = project.addExistingSourceFile(
+      ComponentGenerator.sourceFilePath(component),
+    )
+    const testFile = project.addExistingSourceFile(
+      ComponentGenerator.testFilePath(component),
+    )
+    return { sourceFile, testFile }
+  })
+}
+
+describe("component generator", () => {
+  let compiledGenerator: string
+  let sourceFile: SourceFile
+  let testFile: SourceFile
+
+  const describeExpression = () =>
+    (testFile
+      .getFirstChildOrThrow()
+      .getLastChildOrThrow() as ExpressionStatement).getExpression() as CallExpression
+
+  const testBodySyntaxList = () => {
+    const itExpression = ((describeExpression().getArguments()[1] as ArrowFunction)
+      .getBody()
+      .getChildSyntaxListOrThrow()
+      .getFirstChildOrThrow() as ExpressionStatement).getExpression() as CallExpression
+    return (itExpression.getArguments()[1] as ArrowFunction)
+      .getBody()
+      .getChildSyntaxListOrThrow()
+  }
+
+  beforeAll(() => {
+    compiledGenerator = compileGenerator()
+    expect.hasAssertions()
+  })
+
+  afterAll(() => {
+    fs.unlinkSync(compiledGenerator)
+  })
+
+  it("generates the component and test at the specified location", async () => {
+    ;({ sourceFile, testFile } = await generate(
```

Comment:
`;({ sourceFile, testFile }` so weird - I get what it's doing but that's a funny way for (prettier) to do it, I guess it's the 100% safe from bad ASI

Reply:
It's how each old grumpy man's test should begin: 🤨

Reply:
hah - yep

<img width="685" alt="screen shot 2019-01-02 at 12 50 39 pm" src="https://user-images.githubusercontent.com/49038/50604450-034bfd00-0e8d-11e9-9f2a-7e3175abc899.png">